src/main/java/uk/gov/hmcts/ccd/CachingConfiguration.java

```diff
@@ -22,26 +25,34 @@ public Config hazelCastConfig() {
         networkConfig.getJoin().getMulticastConfig().setEnabled(false);
         networkConfig.getJoin().getTcpIpConfig().setEnabled(false);
-        configCaches(applicationParams.getDefinitionCacheTTLSecs(), config);
+        configCaches(applicationParams.getDefinitionCacheMaxIdleSecs(), applicationParams.getDefinitionCacheTTLSecs(), config);
         return config;
     }
 
-    private void configCaches(int definitionCacheTTL, Config config) {
-        config.addMapConfig(newMapConfig("caseTypeDefinitionsCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("workBasketResultCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("searchResultCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("searchInputDefinitionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("workbasketInputDefinitionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("caseTabCollectionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("wizardPageCollectionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("userRolesCache", definitionCacheTTL));
+    private void configCaches(int definitionCacheMaxIdle, int definitionCacheTTL, Config config) {
+        config.addMapConfig(newMapConfig("caseTypeDefinitionsCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("workBasketResultCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("searchResultCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("searchInputDefinitionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("workbasketInputDefinitionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("caseTabCollectionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("wizardPageCollectionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("userRolesCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("caseTypeDefinitionLatestVersionCache", definitionCacheMaxIdle, definitionCacheTTL));
+        
+    }
+
+    private MapConfig newMapConfig(final String name, Integer definitionCacheMaxIdle) {
+        return newMapConfig(name, definitionCacheMaxIdle, null);
     }
 
-    private MapConfig newMapConfig(final String name, int definitionCacheTTL) {
-        return new MapConfig().setName(name)
+    private MapConfig newMapConfig(final String name, Integer definitionCacheMaxIdle, Integer definitionCacheTTL) {
+        MapConfig mapConfig = new MapConfig().setName(name)
                 .setMaxSizeConfig(new MaxSizeConfig(applicationParams.getDefinitionCacheMaxSize(), MaxSizeConfig.MaxSizePolicy.PER_NODE))
                 .setEvictionPolicy(applicationParams.getDefinitionCacheEvictionPolicy())
-                .setMaxIdleSeconds(definitionCacheTTL);
+                .setMaxIdleSeconds(definitionCacheMaxIdle);
```

Comment:
I think if we use setTimeToLiveSeconds we can/should omit setMaxIdleSeconds, because the 2 serve the same purpose

Reply:
Could this rather stay as it is now? This is because the logic around the 2 parameters are already managed internally in the cache implementation, and if there are 2 facts that apply to the cache, better to provide them and expect the consistent behaviour from the cache design and impl.

Reply:
I think it'll just make things confusing if we keep it both. I don't see the reason. We want the behaviour provided by the ttl and we want to make sure instead we don't get the behaviour of the max idle. So why setting it in the first place? Even if the cache is able to deal with it anyway.

Reply:
Well, I was thinking to design map configs like this: All our definition cache maps have at least  the max-idle defined. And they have an optional value of ttl defined as well. So, max-idle would apply to everything for obvious reason to evict items of no use for a long time, and a particular ttl, when defined, to evict when an items ages enough. This would mean a consistent use of all available parameters for all map configs, and we won't need to remember what we fed into which. Basically, if something is available as a param, it becomes part of the map config it applies to. In this case, I was suggesting max idle to apply to everything, even if the internal cache design/impl makes it logically obsolete.